src/fsharp/SyntaxTrivia.fs

```diff
@@ -4,6 +4,22 @@ namespace FSharp.Compiler.SyntaxTrivia
 
 open FSharp.Compiler.Text
 
+[<RequireQualifiedAccess; NoEquality; NoComparison>]
+type ConditionalDirectiveTrivia =
+    | IfDirectiveTrivia of expr:IfDirectiveExpression * range:range
```

Comment:
Should the tree also contain the evaluation of the ifdef expressions? Or record the conditional defines (eg parsing options) sufficient to make reevaluation easy? Just wondering, it's not essential.

Reply:
We won't need this in our case. Perhaps we can add it later if there is a use case.